Add resource docs and tweak tool source schema title generation

mvdbeek opened the pull request:
Add docs for cores_{min,max}, ram_{min,max} and turn off auto capitalization of field titles for things belonging to the tool source schema.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

mvdbeek committed:
Add more documentation to cores and ram

mvdbeek committed:
Don't auto-capitalize tool source field titles

nsoranzo commented on client/src/components/Tool/ToolSourceSchema.json:
```diff
@@ -1 +1,5 @@
-{"$defs": {"BaseUrlParameterModel": {"properties": {"name": {"description": "Parameter name. Used when referencing parameter in workflows or inside command templating.", "title": "Name", "type": "string"}, "parameter_type": {"const": "gx_baseurl", "default": "gx_baseurl", "title": "Parameter Type", "type": "string"}, "hidden": {"default": false, "title": "Hidden", "type": "boolean"}, "label": {"a ...
+<<<<<<< HEAD
```
Bad merge.

mvdbeek replied:
ah yeah, rebuilt it now

nsoranzo commented on lib/galaxy/tool_util_models/tool_outputs.py:
```diff
@@ -22,6 +22,8 @@
     TypeVar,
```
lint: commands[0]> ruff check .
F401 [*] `pydantic.BaseModel` imported but unused
  --> lib/galaxy/tool_util_models/tool_outputs.py:16:5
   |
15 | from pydantic import (
16 |     BaseModel,
   |     ^^^^^^^^^
17 |     Field,
18 | )
   |
help: Remove unused import: `pydantic.BaseModel`

nsoranzo commented on client/src/api/schema/schema.ts:
```diff
@@ -18968,32 +18972,38 @@ export interface components {
-            cores_max: number | null;
-            /** Cores Min */
+            /** cores_max */
+            cores_max?: number | null;
```
Should the description be shown here?
Same for other resource requirements.

mvdbeek replied:
That's an `openapi-typescript` thing, it is in the full openapi schema. For the suggestions we care about `ToolSourceSchema.json` and it is in there.

nsoranzo replied:
Found the issue, I'll push a commit if that's OK.

mvdbeek replied:
sure

nsoranzo commented on lib/galaxy/tool_util_models/tool_source.py:
```diff
@@ -42,23 +43,41 @@ class SetEnvironmentRequirement(Requirement):
+    shm_size: Optional[Union[int, float]] = None
+
+
+class JavascriptRequirement(ToolSourceBaseModel):
```
Should this inherit from `ResourceRequirement` or we are not interested in specifying cores_min/ram_min/... here any more?

mvdbeek replied:
In JavascriptRquirement ? I don't think that would be right

nsoranzo replied:
Sorry, I misread the diff.

nsoranzo committed:
Move ``Optional[]`` inside ``Annotated[]``

Otherwise title and description are missing from the generated
TypeScript schemas.

nsoranzo committed:
Add type annotations
